From d5f043460ab7592a122c3d8af502dac0d8be4f42 Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Sat, 22 Apr 2023 20:32:02 +0530
Subject: [PATCH 13/31] amd-xgbe : fix to Custom initialization of Marvell PHY
 on Bilby

commit 7d9c4ef190bd7497ec794363e1454b4715fa9c07
Author: rgaridap <Ramesh.Garidapuri@amd.com>
Date:   Tue Apr 11 12:35:30 2023 +0530

    amd-xgbe: Custom initialization of Marvell PHY on Bilby

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
Change-Id: I01d4fb0e4a71e1d10c6fbbefc5a1e1592cacd534
---
 drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
index 9701003af9bb..5eb958b4a7b0 100755
--- a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
+++ b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
@@ -1081,7 +1081,7 @@ static int xgbe_phy_find_phy_device(struct xgbe_prv_data *pdata)
 	}
 
 	ret = phy_attach_direct(pdata->netdev, phydev, phydev->dev_flags,
-				(phy_data->phydev_mode == XGBE_PORT_MODE_BACKPLANE) ? PHY_INTERFACE_MODE_10GKR : PHY_INTERFACE_MODE_SGMII);
+				 PHY_INTERFACE_MODE_SGMII);
 	if (ret) {
 		netdev_err(pdata->netdev, "phy_attach_direct failed\n");
 		phy_device_remove(phydev);
@@ -2176,6 +2176,8 @@ static void xgbe_phy_perform_ratechange(struct xgbe_prv_data *pdata,
 	XP_SET_BITS(s0, XP_DRIVER_SCRATCH_0, COMMAND, cmd);
 	XP_SET_BITS(s0, XP_DRIVER_SCRATCH_0, SUB_COMMAND, sub_cmd);
 
+	netif_dbg(pdata, link, pdata->netdev,"Mailbox CMD  %d , SUBCMD %d\n",cmd,sub_cmd);
+
 	/* Issue the command */
 	XP_IOWRITE(pdata, XP_DRIVER_SCRATCH_0, s0);
 	XP_IOWRITE(pdata, XP_DRIVER_SCRATCH_1, 0);
-- 
2.34.1

